.github/workflows/test-canary.yml

```
@@ -1,5 +1,5 @@
 # This pipeline purpose is solely meant to run a subset of our test suites against upcoming or unreleased dependencies versions
-name: canary
+name: "[see #3988] canary"
```

Maybe add a brief description in a few words

```suggestion
name: "[May fail; see #3988] canary"
```